### PR TITLE
Specify retry duration when DO LB is in a being created ("new" state)

### PR DESCRIPTION
**Pull request by llDrLove**
[Context]
When an LB is being created by CCM, the controller waits for the operation to complete by retrying the create continuously. This leads to an error by the LB API as long as the LB is still being provisioned (with the error from the API saying something along the lines of "LB cannot be updated while create is pending"), which in turn causes CCM to back off at exponentially increasing waiting times. This has the effect that by the time the LB has actually finished provisioning, CCM is in such deep back-off that it takes an unreasonable long time for it to recognize the LB as ready. Anecdotal data indicates that we lose up to a few single-digit minutes until CCM can retrieve the LB IP address and allow the LB to be used in Kubernetes.

Kubernetes 1.28 [added support](https://github.com/kubernetes/kubernetes/pull/94021) for returning a `RetryError` type specifying a custom retry period for CCM to wait. This PR leverages the new error type in our CCM's `EnsureLoadBalancer` method implementation by identifying when an LB is `new` and returning such a type with a relatively short `15` seconds retry time between each reconciliation.

### CODE REVIEW

File: cloud-controller-manager/do/loadbalancers.go [hunk 1]
```diff
@@ -218,6 +220,10 @@ func (l *loadBalancers) EnsureLoadBalancer(ctx context.Context, clusterName stri
 	}
 
 	if lb.Status != lbStatusActive {
+		if lb.Status == lbStatusNew {
```
Comment (timoreimann) [inline cloud-controller-manager/do/loadbalancers.go:223]: I'd move the new `if` statement outside of the pre-existing `if` statement to ease readability.

File: cloud-controller-manager/do/loadbalancers.go [hunk 2]
```diff
@@ -218,6 +220,10 @@ func (l *loadBalancers) EnsureLoadBalancer(ctx context.Context, clusterName stri
...
+			return nil, api.NewRetryError("load-balancer is currently being created", 15*time.Second)
+		}
+
 		return nil, fmt.Errorf("load-balancer is not yet active (current status: %s)", lb.Status)
```
Comment (timoreimann) [inline cloud-controller-manager/do/loadbalancers.go:227]: I'd rephrase this now that any status other than "active" is unexpected and treated as a genuine error:

```suggestion
		return nil, fmt.Errorf("load-balancer has unexpected status %q", lb.Status)
```

**Review by timoreimann — COMMENTED** [review]
This looks good, just two small improvement suggestions.

File: cloud-controller-manager/do/loadbalancers.go (hunk 1 above)
Comment (llDrLove) [reply cloud-controller-manager/do/loadbalancers.go:223]: Sure, I've moved it above the existing `if`.

File: cloud-controller-manager/do/loadbalancers.go (hunk 2 above)
Comment (llDrLove) [reply cloud-controller-manager/do/loadbalancers.go]: Addressed.

**Comment by llDrLove** [comment]
Added the Changelog entry.
